### PR TITLE
Add Loader.loadBinary: wasm modules will need to be fetched in binary format

### PR DESCRIPTION
**Pull request by mykmartin**

### CODE REVIEW

File: src/runtime/loader.ts [hunk 1]
```diff
@@ -63,16 +63,24 @@ export class Loader {
...
-  async _loadFile(file: string): Promise<string> {
+  async loadBinary(file: string): Promise<ArrayBuffer> {
+    if (/^https?:\/\//.test(file)) {
+      return fetch(file).then(res => res.arrayBuffer());
+    } else {
+      return this._loadFile(file, null) as Promise<ArrayBuffer>;
+    }
+  }
+
+  async _loadFile(file: string, encoding?: string): Promise<string | ArrayBuffer> {
```
Comment (piotrswigon) [inline src/runtime/loader.ts]: I recently learned you can do something like:

```
interface Mapping {
  'utf-8': string;
  'binary': ArrayBuffer;
}
async _loadFile<T extends keyof Mapping>(file: string, encoding: T): Promise<Mapping[T]> {
  ...
}
```

File: src/runtime/manual_tests/loader-tests.ts [hunk 2]
```diff
@@ -9,29 +9,69 @@
  */
 
 import {assert} from '../../platform/chai-web.js';
+import {fs} from '../../platform/fs-web.js';
+import {path} from '../../platform/path-web.js';
 import {Loader} from '../loader.js';
 import {Manifest} from '../manifest.js';
 
-describe('loader', () => {
+describe.only('loader', function() {
```
Comment (piotrswigon) [inline src/runtime/manual_tests/loader-tests.ts]: only?

Comment (mykmartin) [reply src/runtime/manual_tests/loader-tests.ts]: Yep, fixed in a follow-up.

File: src/runtime/loader.ts (hunk 1 above)
Comment (mykmartin) [reply src/runtime/loader.ts]: Does that work for a null input? `encoding` is actually either 'utf-8' or null.

... but even if it does, I'm not sure that's better. The API itself is a lot less readable IMO.

Comment (piotrswigon) [reply src/runtime/loader.ts]: It does work with null. I would do it for type safety and getting rid of casting and all that.

I agree it's a bit verbose, but I would still take it for type safety.

Use your best judgement.

**Pull request merged**